src/Core/Checkout/Cart/Facade/ContainerFacade.php

```diff
@@ -77,10 +72,10 @@ public function add(ItemFacade $item): ItemFacade
     {
         $this->item->getChildren()->add($item->getItem());
 
-        /** @var ItemFacade $item */
-        $item = $this->get($item->getId());
+        $addedItem = $this->get($item->getId());
+        \assert($addedItem instanceof ItemFacade, 'Item was just added, so it should be available');
```

Comment:
What is the point of this code? If it's just for fluent interface why not just return the item directly instead of fetching it again?

Reply:
I still don't really see why that is necessary, it would just manipulate the same object, but it's fine as it was already like that 👍

Reply:
:thinking: you are right, I always forget about that reference stuff :sweat_smile: